File: server/src/utils/filenameEncoding.ts
export function normalizePossibleMojibake(value: string): string {
  try {
    const decoded = Buffer.from(value, "latin1").toString("utf8");
    const roundTrip = Buffer.from(decoded, "utf8").toString("latin1");

    return roundTrip === value ? decoded : value;
  } catch {
    return value;
  }
}

File: backend/src/utils/filename_encoding.py
def normalize_possible_mojibake(value: str) -> str:
    if not isinstance(value, str):
        return value

    try:
        decoded = value.encode("latin1").decode("utf-8")
        round_trip = decoded.encode("utf-8").decode("latin1")
        return decoded if round_trip == value else value
    except (UnicodeEncodeError, UnicodeDecodeError):
        return value
